pkg/target/target.go

```diff
@@ -57,22 +62,24 @@ type Target interface {
 }
 
 type targetImpl struct {
-	Garden  string `yaml:"garden,omitempty" json:"garden,omitempty"`
-	Project string `yaml:"project,omitempty" json:"project,omitempty"`
-	Seed    string `yaml:"seed,omitempty" json:"seed,omitempty"`
-	Shoot   string `yaml:"shoot,omitempty" json:"shoot,omitempty"`
+	Garden       string `yaml:"garden,omitempty" json:"garden,omitempty"`
+	Project      string `yaml:"project,omitempty" json:"project,omitempty"`
+	Seed         string `yaml:"seed,omitempty" json:"seed,omitempty"`
+	Shoot        string `yaml:"shoot,omitempty" json:"shoot,omitempty"`
+	ControlPlane bool   `yaml:"controlPlane,omitempty" json:"controlPlane,omitempty"`
 }
 
 var _ Target = &targetImpl{}
 
 // NewTarget returns a new target. This function does not perform any validation,
 // so the returned target can be invalid.
-func NewTarget(gardenName, projectName, seedName, shootName string) Target {
+func NewTarget(gardenName, projectName, seedName, shootName string, shootControlPlane bool) Target {
```

Comment:
Note (just for me to remember if this question comes up) - I wanted to add a check to the `Validate()` method, something like
```
if t.ControlPlane && len(t.Shoot) == 0 {
    return errors.New("control plane can only be configured if a shoot is targeted")
}
```
However, this was not possible as this would prevent the user from targeting a shoot control plane like this:
`gardenctl target shoot foo --controlplane`
As the shoot is not set on the `currenttarget` at the point of `Validate()` function invocation.

There will be an error thrown by the `targetBuilder` anyway when trying to target a control plane without targeting a shoot before or along with the command.